src/Config/EnvironmentData.php

```diff
@@ -114,18 +136,31 @@ public function getApplication(): array
             return $this->data['application'];
         }
 
-        return $this->data['application'] = $this->getEnvVar(SystemConfigInterface::VAR_ENV_APPLICATION, []);
+        $application = $this->getEnvVar(SystemConfigInterface::VAR_ENV_APPLICATION, []);
+
+        if (!$application) {
+            // Try to read from .magento.app.yaml
+            try {
+                $application = Yaml::parse(
+                    $this->file->fileGetContents($this->fileList->getAppConfig())
+                );
+            } catch (FileSystemException $e) {
+                // Do nothing as $application needs to be empty
```

Comment:
Instead of a comment, I think better to add some warning message about neither env variable and .magento.app.yaml  not exists

Reply:
But I'm not sure about the message, @YPyltiai what do you think?

Reply:
Well, I left it as nothing as that is the current behaviour. I did think about putting a message in but left it. If the consensus is that at message is needed, should it be output direct to console line or done as a log? I'd lean towards the former as in theory it should only happen on a local instance if the file is missing.

Reply:
I've discussed with @YPyltiai and we think that message is not needed. Let's leave as it is, thank you.

Reply:
Okay - I made the other changes you requested so can you approve the PR?
BTW I don't understand why the CI Checks are failing. The failure doesn't seem appropriate ?
@oshmyheliuk @YPyltiai

Reply:
Yes, sorry, I thought that I've already approved